pkg/apiserver/resthandler.go

```diff
@@ -147,7 +147,20 @@ func getRequestOptions(req *restful.Request, scope RequestScope, kind string, su
 		newQuery[subpathKey] = []string{req.PathParameter("path")}
 		query = newQuery
 	}
-	return queryToObject(query, scope, kind)
+	versioned, err := scope.Creater.New(scope.ServerAPIVersion, kind)
+	if err != nil {
+		// programmer error
+		return nil, err
+	}
+	if err := scope.Codec.DecodeParametersInto(query, versioned); err != nil {
```

Comment:
Yes, also, as @liggitt says, query parameters do *not* contain their own type and therefore can't be auto-converted.